### ISSUE TITLE
False positive: 'No overload matches this call' when using snippet syntax

### ISSUE THREAD

**Issue by janosh** [pheuter/svelte-check-rs#90]
When using Svelte 5's snippet syntax to pass content to a component, svelte-check-rs reports "No overload matches this call" errors.

## Reproduction

**Parent component (Parent.svelte):**
```svelte
<script lang="ts">
  import type { Snippet } from 'svelte'

  let { children, inset }: { children?: Snippet; inset?: Snippet } = $props()
</script>

{#if children}
  {@render children()}
{/if}
{#if inset}
  {@render inset()}
{/if}
```

**Usage:**
```svelte
<script lang="ts">
  import Parent from './Parent.svelte'
</script>

<Parent>
  {#snippet inset()}
    <div>Inset content</div>
  {/snippet}
</Parent>
```

## Error

```
Error: No overload matches this call. (ts(TS2769))
```

svelte-check-rs: 0.7.1
